stylix/nixos/default.nix

```diff
@@ -0,0 +1,25 @@
+{ palette-generator, base16 }:
+{ config, lib, ... }:
+
+let
+  cfg = config.stylix.home-manager;
+  autoload = import ../autoload.nix { inherit lib; } "nixos";
+in {
+  imports = [
+    ../pixel.nix
+    ../target.nix
+    ./fonts.nix
+    (import ./palette.nix { inherit palette-generator base16; })
+  ] ++ autoload;
+
+  options.stylix.home-manager = {
+    enable = lib.mkEnableOption "home-manager support";
```

Comment:
Am I right in thinking this is disabled by default?

If so, it should either be enabled automatically when Home Manager is present, or output a warning message.

This is just for backwards compatibility, as some users will update without knowing about these changes, and may be confused if all of the Home Manager based modules stop working.

Reply:
My last commit should fix that. I also added a way to disable the import of the hm module, because if it is added manually by the user, the double inclusion will cause conflict.